refactor(rust): add `NodeManagerWorker` which handles the requests

michealkeines opened the pull request:
Test version of suggested solutions in https://github.com/build-trust/ockam/issues/3524

1. Rename NodeManager to NodeManagerWorker and let it only handle requests
2. Move all state to a separate structure (name NodeManager is still good here)

I have added the `NodeManagerWorker` which will handle the requests, i wasnt sure about which states to keep in the `NodeManager`, @SanjoDeundiak please take a look a these changes and let me know what else should be added Thankyou.

## Checks

<!-- To help us review and merge this pull request quickly, please confirm the following:  -->

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

<!-- Looking forward to merging your contribution!! -->

adrianbenavides reviewed (commented):
@michealkeines thanks for this PR! This is going to make things a lot easier on the `NodeManager` side :+1: 

I've tried to force-push on your branch to fix the lint error but I didn't have access. The PR will be ready to merge as soon as you fix a couple of things:
1. There is a small linting error that can be fixed by running `cargo fmt --all`
2. You should also rebase the latest develop changes to your branch: 
```
git checkout develop
git pull --rebase
git checkout issue-3524-create-node-manger-worker
git rebase develop
```

@SanjoDeundiak let me know if the PR is good to merge on your side :pray:

michealkeines committed:
refactor(rust): add `NodeManagerWorker` which handles the requests

SanjoDeundiak commented:
Hey @michealkeines . Nice PR! Let's merge it!
Also, I think the initial issue involves more changes, specifically, in case you want to continue working on that issue here what I would suggest:
1. Move `handle_request` function to the `NodeManagerWorker`
2. Move functions that do some Request/Response handling from `NodeManager` to `NodeManagerWorker`. Those are usually the ones that are called from `handle_request`. Implementation functions (sometimes end with `_impl`) should stay in `NodeManager`
3.  Make `node_manager` field of `NodeManagerWorker` private, create a getter for it and a constructor for `NodeManagerWorker`

The goal of all of that is better separation between the `NodeManager` as a node's state owner, and the `NodeManagerWorker` as a worker serving some functionality that someone connected to our node can perform.

michealkeines commented:
@SanjoDeundiak, Thank you for the feedback, i will check the things you have mentioned and will continue working on this issue.

michealkeines committed:
refactor(rust): move methods into `NodeManagerWorker`

1. moved `handle_request` from NodeManger to NodeManagerWorker
2. moved Request/Response handling methods to NodeManagerWorker (all the
   methods that were called inside `handle_request`)
3. Get and Constructor for NodeManagerWorker struct

michealkeines commented:
Hi @SanjoDeundiak I have made the suggested changes

1. moved `handle_request` from NodeManger to NodeManagerWorker
2. moved Request/Response handling methods to NodeManagerWorker (all the methods that were called inside `handle_request`)
3. Get and Constructor for NodeManagerWorker struct

Please review, Thank you
